lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskRuntime.cs

```diff
@@ -31,58 +31,41 @@
 
 namespace Org.Apache.REEF.Common.Runtime.Evaluator.Task
 {
-    public class TaskRuntime : IObserver<ICloseEvent>, IObserver<ISuspendEvent>, IObserver<IDriverMessage>
+    internal sealed class TaskRuntime : IObserver<ICloseEvent>, IObserver<ISuspendEvent>, IObserver<IDriverMessage>
     {
         private static readonly Logger LOGGER = Logger.GetLogger(typeof(TaskRuntime));
-        
-        private readonly ITask _task;
 
         private readonly IInjector _injector;
 
-        // The memento given by the task configuration
-        private readonly Optional<byte[]> _memento;
-
-        private readonly HeartBeatManager _heartBeatManager;
-
         private readonly TaskStatus _currentStatus;
 
-        private readonly INameClient _nameClient;
-
         private readonly Lazy<IDriverConnectionMessageHandler> _driverConnectionMessageHandler;
 
-        public TaskRuntime(IInjector taskInjector, string contextId, string taskId, HeartBeatManager heartBeatManager, string memento = null)
```

Comment:
I thought ``memento`` was something that was meant to be deprecated in favor of Tang injection. Perhaps instead of fixing the C# code, we should be deprecating the Java function call instead. @markusweimer can you confirm?